packages/viewer/src/store/modules/layers/actions/identifyFeatures.ts

```diff
@@ -0,0 +1,80 @@
+import type { Layer } from '@swissgeo/layers'
+
+import type { LayersStore } from '@/store/modules/layers/types/layers'
+import type { ActionDispatcher } from '@/store/types'
+
+import useFeaturesStore from '@/store/modules/features'
+import useMapStore from '@/store/modules/map'
+
+export interface GetLayerIdOptions {
+    layerOrIndex?: number | string | Layer
+    index?: number
+    activeLayer?: Layer
+}
+export interface GetLayerIdResult {
+    layerId?: string
+    updateFeatures?: boolean
+}
+
+export function identifyFeatures(
+    this: LayersStore,
+    dispatcher: ActionDispatcher
+): void
+export function identifyFeatures(
+    this: LayersStore,
+    getLayerContext: (options: GetLayerIdOptions) => GetLayerIdResult,
+    options: GetLayerIdOptions,
+    dispatcher: ActionDispatcher
+): void
+export function identifyFeatures(
+    this: LayersStore,
+    functionOrDispatcher: ((options: GetLayerIdOptions) => GetLayerIdResult) | ActionDispatcher,
+    optionsOrDispatcher?: GetLayerIdOptions | ActionDispatcher,
+    dispatcherOrNothing?: ActionDispatcher
+): void {
+    // Determine which overload was called
+    let getLayerContext: ((options: GetLayerIdOptions) => GetLayerIdResult) | undefined
+    let options: GetLayerIdOptions = {}
+    let dispatcher: ActionDispatcher
+
+    if (typeof functionOrDispatcher === 'function' && optionsOrDispatcher && dispatcherOrNothing) {
+        // Second overload: getLayerContext, options, dispatcher
+        getLayerContext = functionOrDispatcher as (options: GetLayerIdOptions) => GetLayerIdResult
+        options = optionsOrDispatcher as GetLayerIdOptions
+        dispatcher = dispatcherOrNothing
+    } else {
+        // First overload: dispatcher only
+        getLayerContext = undefined
+        options = {}
+        dispatcher = functionOrDispatcher as ActionDispatcher
+    }
+
+    const featuresStore = useFeaturesStore()
+    const mapStore = useMapStore()
+    const selectedFeatures = featuresStore.selectedFeatures
+    let clickInfo = mapStore.clickInfo
+
+    if (!this.isFeatureSelected(clickInfo)) {
+        return
+    }
+    clickInfo = clickInfo!
+
+    // for 'setLayerYear', 'addLayer', 'clearLayers', 'removeLayerByIndex' we always update
+    const { layerId, updateFeatures: shouldUpdateFeatures = true } = getLayerContext?.(options) ?? {}
+    let updateFeatures = shouldUpdateFeatures
+
+    if (layerId) {
+        updateFeatures = selectedFeatures.some(
+            (feature) => 'layer' in feature && feature.layer.id === layerId
+        )
+    }
+
+    if (updateFeatures && clickInfo.features) {
```

Comment:
so without a `clickInfo.feature` there's nothing to do?
Might be also a good idea to add a `|| !clickInfo.features` case to the `if` guarding block with the early return then